docs/basics/101-130-yodaproject.rst

```diff
@@ -0,0 +1,564 @@
+   :command:`datalad publish` to this sibling, your datasets history
+   will be pushed there.
+
+Yay! Consider your midterm project submitted! Others can now install your
+dataset and check out your data science project -- and even better: they can
+reproduce your data science project easily from scratch!
+
+.. findoutmore:: On the looks and feels of this published dataset
+
+   Now that you have created and published such a YODA-compliant dataset, you
+   are understandably excited how this dataset must look and feel for others.
+   Therefore, you decide to install this dataset in a new location on your
+   computer, just to get a feel for it.
+
+   Replace the ``url`` in the :command:`install` command below with the path
+   to your own ``midtermproject`` Github repository:
+
+   .. runrecord:: _examples/DL-101-130-125
+      :language: console
+      :workdir: dl-101/DataLad-101/midterm_project
+
+      $ cd ../../
+      $ datalad install -r --source "https://github.com/adswa/midtermproject.git"
+
+   Note that we performed a *recursive* installation by providing the ``-r``
+   option. Thus, we don't need to install the ``input/`` subdataset anymore.
+   Let's start with the subdataset, and see whether we can retrieve the
+   input ``iris.csv`` file. This should not be a problem, since it's origin
+   is recorded:
+
+
+   .. runrecord:: _examples/DL-101-130-126
+      :language: console
+      :workdir: dl-101
+
+      $ cd midtermproject
+      $ datalad get input/iris.csv
+
+   Nice, this worked well. The output files, however, can not be easily
+   retrieved:
+
+   .. runrecord:: _examples/DL-101-130-127
+      :language: console
+      :workdir: dl-101/midtermproject
+
+      $ datalad get output/*
+
+   Why is that? The file content of these files is managed by Git-annex, and
+   thus only information about the file name and location is known to Git.
+   Because Github does not host large data, annexed file content always
+   needs to be deposited somewhere else (e.g., a webserver) to make it
+   accessible via :command:`datalad get`. A later section
+
+   .. todo::
+
+      link 3rd party infra section
+
+   will demonstrate how this can be done. For this dataset, it is not
+   necessary to make the outputs available, though: Because all provenance
+   on their creation was captured, we can simply recompute them with the
+   :command:`datalad rerun` command.
+
+   .. runrecord:: _examples/DL-101-130-128
+      :language: console
+      :workdir: dl-101/midtermproject
+      :realcommand: echo "datalad rerun $(git rev-parse HEAD~2)" && datalad rerun $(git rev-parse HEAD~2)
+
+   Hooray, your analysis was reproduced! Note however that none of the DataLad
+   commands would have been necessary to reproduce the analysis -- simply
+   executing the Python script would have installed and retrieved the inputs thanks
+   to the DataLad functions used in the script, as long as DataLad is installed
+   on the system.
```

Comment:
This is a bit confusing. DataLad is not needed, but DataLad needs to be installed. I think it would be good to emphasize that the rerunning part did not even require knowledge about the analysis. It might be possible to factor out the `HEAD~2` as well, and just use a tag. It would look even more generic then.